### PR TITLE
Fix HMR - Kill child process only on first build

### PR DESCRIPTION
**Pull request by sadafie**
After a `yarn run start:full` or `yarn run start`, if you encounter a build error in the WDS, we don't attempt to kill the process. However, if it's the first build, we kill all child processes and the main node process as well.

This works great in the HMR case. However, If you have an error on the first build, it works but with a weird side-effect:

```sadaf: ~/git_repo/mup-web [feature/upgrade_mwp_cli_may13] [!] $ yarn run start
yarn run v1.6.0
$ mope run
NODE_ENV=development
>> start-dev: Preparing the Dev App Server using existing vendor bundle...
(node:41424) DeprecationWarning: Tapable.plugin is deprecated. Use new API on `.hooks` instead
>> ./src/assets/scss/main.scss (./node_modules/file-loader/dist/cjs.js?name=[name].[hash:7].css!./node_modules/extract-loader/lib/extractLoader.js!./node_modules/css-loader!./node_modules/sass-loader/lib/loader.js!./src/assets/scss/main.scss)
   Module build failed:
         in src/assets/scss/main.scss (line 141, column 6)
   Error:
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
sadaf: ~/git_repo/mup-web [feature/upgrade_mwp_cli_may13] [!] $ events.js:165
      throw er; // Unhandled 'error' event
      ^

Error [ERR_IPC_CHANNEL_CLOSED]: Channel closed
    at process.target.send (internal/child_process.js:599:16)
    at compiler.plugin.stats (/Users/sadaf/git_repo/mup-web/node_modules/mwp-cli/src/commands/runScripts/_webpack-dev-server.js:12:43)
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/Users/sadaf/git_repo/mup-web/node_modules/tapable/lib/HookCodeFactory.js:24:12), <anonymous>:9:1)
    at AsyncSeriesHook.lazyCompileHook [as _callAsync] (/Users/sadaf/git_repo/mup-web/node_modules/tapable/lib/Hook.js:35:21)
    at Watching._done (/Users/sadaf/git_repo/mup-web/node_modules/webpack/lib/Watching.js:99:28)
    at compiler.emitRecords.err (/Users/sadaf/git_repo/mup-web/node_modules/webpack/lib/Watching.js:73:19)
    at Compiler.emitRecords (/Users/sadaf/git_repo/mup-web/node_modules/webpack/lib/Compiler.js:315:39)
    at compiler.emitAssets.err (/Users/sadaf/git_repo/mup-web/node_modules/webpack/lib/Watching.js:54:20)
    at hooks.afterEmit.callAsync.err (/Users/sadaf/git_repo/mup-web/node_modules/webpack/lib/Compiler.js:301:14)
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/Users/sadaf/git_repo/mup-web/node_modules/tapable/lib/HookCodeFactory.js:24:12), <anonymous>:6:1)
Emitted 'error' event at:
    at processEmit [as emit] (/Users/sadaf/git_repo/mup-web/node_modules/signal-exit/index.js:155:32)
    at process.nextTick (internal/child_process.js:603:35)
    at process._tickCallback (internal/process/next_tick.js:176:11)
```

It error and kills the process correctly, however, after about a 10sec delay it emits the second error:

```sadaf: ~/git_repo/mup-web [feature/upgrade_mwp_cli_may13] [!] $ events.js:165
      throw er; // Unhandled 'error' event
      ^

Error [ERR_IPC_CHANNEL_CLOSED]: Channel closed
    at process.target.send (internal/child_process.js:599:16)
```

That you have to `Ctrl+C` interrupt to quit. I tried adding a few `wdsProcess.on('error')` error handling, but unfortunately that didn't work. The uncaught error doesn't actually cause any issues. You can exit it and move on, but it's an annoying error that I'd like to avoid printing out.

### CODE REVIEW

File: src/commands/runScripts/start-dev.js [hunk 1]
```diff
@@ -108,12 +126,12 @@ function run() {
...
 	 */
 	const browserAppCompileLogger = getCompileLogger('browserApp');
-	const wdsProcess = fork(path.resolve(__dirname, '_webpack-dev-server'));
+	wdsProcess = fork(path.resolve(__dirname, '_webpack-dev-server'));
 
 	// the dev server compiler will send a message each time it completes a build
 	wdsProcess.on('message', message => {
 		browserAppCompileLogger();
-		if (!ready.browerApp) {
+		if (!ready.browserApp) {
```
Comment (sadafie) [inline src/commands/runScripts/start-dev.js]: Oops, found a typo!

**Comment by mmcgahan** [comment]
Just a guess, but maybe you should remove the `'data'` event listener before killing the webpack process? https://nodejs.org/api/events.html#events_emitter_removealllisteners_eventname

**Comment by sadafie** [comment]
So, interesting enough, I noticed I'm on node v 9.11.1, so I downgraded to LTS 8.11.1, and I don't run into this issue anymore. Doing some googling, I'm noticing issues filed under other repos that seem to point to specific minor versions of node 9 that cause this issue. Do you think it's safe to merge this for now?

**Comment by mmcgahan** [comment]
As long as there are docs for what to do when you see certain errors, fine to merge - if we expect concerns/questions, I just want to makes sure we can point people in the right direction

**Comment by mmcgahan** [comment]
would also be interestin to try on Node 10

**Comment by sadafie** [comment]
Added a doc here - https://meetup.atlassian.net/wiki/spaces/WP/pages/569147642/Resolve+mwp-cli+Emitted+error+event

**Pull request merged**